components/aks-mis/demo.tf

```diff
@@ -0,0 +1,35 @@
+data "azurerm_user_assigned_identity" "sops_demo_mi" {
+  name                = "aks-${var.environment}-mi"
+  resource_group_name = "genesis-rg"
+}
+
+resource "azurerm_role_assignment" "demo_externaldns_read_rg" {
+  count                = (contains(["demo"], var.environment) ? 1 : 0) * var.cluster_count
+  provider             = azurerm.dts-cftptl-intsvc
+  principal_id         = data.azurerm_user_assigned_identity.sops_demo_mi.principal_id
+  scope                = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/core-infra-intsvc-rg"
+  role_definition_name = "Reader"
+}
+resource "azurerm_role_assignment" "demo_externaldns_private_dns_zone_contributor" {
+  count                = (contains(["demo"], var.environment) ? 1 : 0) * var.cluster_count
```

Comment:
what's this got to do with cluster count?

Reply:
[e66d624](https://github.com/hmcts/aks-cft-deploy/pull/273/commits/e66d6248122a6a67838b63529bf59476cc93fd00) removed here @timja